python/tests/backends/test_IQM.py

```diff
@@ -44,9 +44,15 @@ def startUpMockServer():
 
     # Set the targeted QPU
     os.environ["IQM_TOKENS_FILE"] = tmp_tokens_file.name
-    cudaq.set_target("iqm",
-                     url="http://localhost:{}".format(port),
-                     **{"qpu-architecture": "Apollo"})
+    with os.popen("git rev-parse --show-toplevel") as f:
+        git_top = f.read().strip()
+    cudaq.set_target(
+        "iqm",
+        url="http://localhost:{}".format(port),
+        **{
+            "qpu-architecture": "Apollo",
+            "mapping-file": f"{git_top}/test/Supplemental/Apollo Variant.txt"
```

Comment:
If we convert these to underscores instead of dashes, wouldn't we get the nicer syntax 
```python
cudaq.set_target('iqm', url='..', qpu_architecture='Apollo', mapping_file='...')
```

Reply:
Yes, good point. Updated in 49d800a55.

Reply:
Oh, you probably meant the `qpu-architecture`, too. Let me take a look.

Reply:
> Oh, you probably meant the `qpu-architecture`, too. Let me take a look.

Updated in 33c69a329. In order to not be a breaking change, I had to make sure the old way still works, too. I still updated the docs to reflect the new way.

Reply:
On second thought, since the `qpu-architecture` change affects the user interface, it feels like that change should be done under a different PR. I will back it out of this PR, and once this one is completed, I will post a separate PR with a more appropriate title to reflect the change (and probably beef up the docs to say the prior way is deprecated).